src/transport/RendezvousSession.cpp

```diff
@@ -351,8 +352,9 @@ CHIP_ERROR RendezvousSession::HandleSecureMessage(PacketBuffer * msgBuf)
 #endif
     plainText = msgBuf->Start();
 
-    // TODO: We need length checks here!  https://github.com/project-chip/connectedhomeip/issues/2928
-    err = mac.Decode(packetHeader, &data[packetHeader.GetPayloadLength()], kMaxTagLen, &taglen);
+    payloadlen = packetHeader.GetPayloadLength();
+    VerifyOrExit(payloadlen < len, err = CHIP_ERROR_INCORRECT_STATE);
```

Comment:
Error code could be changed.

Reply:
Happy to do that; what would you prefer?

Reply:
https://github.com/project-chip/connectedhomeip/pull/3315 will use a better code here.